src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterFactory.cs

```diff
@@ -305,7 +309,11 @@ private EventHubAdapterReceiver MakeReceiver(QueueId queueId)
                 this.EventHubReceiverFactory);
         }
 
-        private async Task<string[]> GetPartitionIdsAsync()
+        /// <summary>
+        /// Get partition Ids from eventhub
+        /// </summary>
+        /// <returns></returns>
+        protected virtual async Task<string[]> GetPartitionIdsAsync()
```

Comment:
Need to either make GetPartitionIdsAsync virtual or CreateAdapter virtual so that EventDataGeneratorStreamProvider won't connect event hub and ask for partition info, but instead use our mocked partition ids.